packages/toolpad-app/src/components/propertyControls/SelectOptions.tsx

```diff
@@ -0,0 +1,226 @@
+            </DialogTitle>
+            <DialogContent>
+              <Stack gap={1} py={1}>
+                <TextField
+                  label="Value"
+                  size="small"
+                  value={editingOption.value}
+                  onChange={(event) => {
+                    handleOptionChange({ ...editingOption, value: event.target.value });
+                  }}
+                />
+                <TextField
+                  label="Label"
+                  size="small"
+                  value={editingOption.label}
+                  onChange={(event) => {
+                    handleOptionChange({ ...editingOption, label: event.target.value });
+                  }}
+                />
+              </Stack>
+            </DialogContent>
+          </React.Fragment>
+        ) : (
+          <React.Fragment>
+            <DialogTitle>
+              Edit options
+              {value.length > 0 ? (
+                <Button
+                  aria-label="Delete all options"
+                  variant="text"
+                  size="small"
+                  color="inherit"
+                  onClick={handleDeleteAll}
+                  sx={{
+                    position: 'absolute',
+                    right: 16,
+                    top: 16,
+                  }}
+                >
+                  Delete All
+                </Button>
+              ) : null}
+            </DialogTitle>
+            <DialogContent>
+              {value.length > 0 ? (
+                <List dense>
+                  {value.map((option, i) => {
+                    return (
+                      <ListItem
+                        key={i}
+                        disableGutters
+                        onClick={handleOptionItemClick(i)}
+                        secondaryAction={
+                          <IconButton
+                            aria-label="Delete option"
+                            edge="end"
+                            onClick={handleOptionDelete(i)}
+                          >
+                            <DeleteIcon fontSize="small" />
+                          </IconButton>
+                        }
+                      >
+                        <ListItemButton>
+                          <ListItemText
+                            primary={
+                              typeof option === 'string' ? option : (option as SelectOption).value
+                            }
+                            secondary={
+                              typeof option === 'object'
+                                ? `Label: "${(option as SelectOption).label}"`
+                                : null
+                            }
+                          />
+                        </ListItemButton>
+                      </ListItem>
+                    );
+                  })}
+                </List>
+              ) : null}
+              <FormControl fullWidth sx={{ m: 1 }} size="small" variant="outlined">
```

Comment:
I'm only superficially reviewing this, but what does this implementation offer over `<TextField variant='outlined' />`?

Reply:
On second look, nothing really - refactored to use just `TextField`